django_components/management/commands/startcomponent.py

```diff
@@ -0,0 +1,165 @@
+import os
+from textwrap import dedent
+
+from django.conf import settings
+from django.core.management.base import BaseCommand, CommandError
+
+
+class Command(BaseCommand):
+    help = "Creates a new component"
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "name", type=str, help="The name of the component to create"
+        )
+        parser.add_argument(
+            "--path",
+            type=str,
+            help="The path to the components directory",
+            default=None,
+        )
+        parser.add_argument(
+            "--js",
+            type=str,
+            help="The name of the javascript file",
+            default="script.js",
+        )
+        parser.add_argument(
+            "--css",
+            type=str,
+            help="The name of the style file",
+            default="style.css",
+        )
+        parser.add_argument(
+            "--template",
+            type=str,
+            help="The name of the template file",
+            default="template.html",
+        )
+        parser.add_argument(
+            "--force",
+            action="store_true",
+            help="Overwrite existing files if they exist",
+        )
+        parser.add_argument(
+            "--verbose",
+            action="store_true",
+            help="Print additional information during component creation",
+        )
+        parser.add_argument(
+            "--dry-run",
+            action="store_true",
+            help="Simulate component creation without actually creating any files",
+            default=False,
+        )
+
+    def handle(self, *args, **kwargs):
+        name = kwargs["name"]
+
+        if name:
+            path = kwargs["path"]
+            js_filename = kwargs["js"]
+            css_filename = kwargs["css"]
+            template_filename = kwargs["template"]
+            base_dir = settings.BASE_DIR
```

Comment:
Isn't BASE_DIR a non-standard variable? Are we sure this one always exist, and what do we do if it doesn't exist? Is this a new requirement of django-components?

Reply:
It exists on the Django default setup.

How should we calculate the base directory of the project? What's your suggestion?

Reply:
@hanifbirgani Sorry for the late reply. Let's check for BASE_DIR, then fall back to PROJECT_PATH (seems semi-standard too), and otherwise error out?

Reply:
Do you mean `PROJECT_PATH` in the environment variables? like `os.environ['PROJECT_PATH']`?

Reply:
No, I've seen some project have a settings.PROJECT_PATH instead of BASE_DIR, and just thought this might be a good fallback if it exists?

Reply:
IMO it's a rabbit hole. I prefer to check the standard `settings.BASE_DIR` and if it doesn't exist, ask the user to specify the full path to create the component. We already have the command line argument (`--path`) for this.

Reply:
Sure. But maybe then some error handling when BASE_DIR doesn't exist?

Reply:
Sure, I'll add error handling in the next commit.

Reply:
@EmilStenstrom I fixed this issue in one of the last commits. Please check the results.